### DISCUSSION TITLE
Step Walkthrough

### DISCUSSION THREAD

**Discussion by twierman** [TheCodeTraveler/XamList#51] [Q&A]
The current step walkthrough is not very compatible with the current Azure interface.

In step 6, you say to select SQLDatabase from the dropdown.  The dropdown does not contain that option:
![2022-01-10_09h25_40](https://user-images.githubusercontent.com/8051522/148801516-90134d0e-3764-46b1-86ff-1472183188ff.png)

When I publish the function app, VS says it was successful.  Azure says that it cannot find:

Microsoft.EntityFrameworkCore: Could not load file or assembly 'Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.

I have tried to go through the walkthrough at least four times and cannot get anything to work.  I am unsure if that error existed previously.

In step 11, neither AzureFunctionKey_RemoveItemFromDatabase or AzureFunctionKey_RestoreDeletedContacts exist in BackEndConstants.cs or anywhere in the project.

When I run the App, I get an error in the ContactDetailPage because finding the database fails.  The 
 async void HandleSaveContactCompleted(object sender, bool isSaveSuccessful)
        {
            if (isSaveSuccessful)
                await PopPage();
           // else
                //await DisplayAlert("Save Failed", "", "OK");
        }
gives me a threading error for the DisplayAlert section.

Albert Moriollo said that I should contact you via Twitter.  You have not responded.  That's OK, since I do not do Twitter.

**Comment by TheCodeTraveler** [comment]
Hi @twierman!

I created this sample, and its walkthrough, about five years ago. It's not surprising that the screenshots are now out of date. 

If you'd like to open a pull request to update it, I'd be happy to merge it!

**Comment by twierman** [comment]
I do not know how to fix it.  I was hoping you could.

On Mon, Jan 10, 2022 at 9:46 AM Brandon Minnick ***@***.***>
wrote:

> Hi @twierman <https://github.com/twierman>!
>
> I created this sample, and its walkthrough, about five years ago. It's not
> surprising that the screenshots are now out of date.
>
> If you'd like to open a pull request to update it, I'd be happy to merge
> it!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/brminnick/XamList/issues/50#issuecomment-1009102630>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB5NWQRA56NLI7OF2ABGS6DUVMEOLANCNFSM5LUDTRQA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Tom Wierman*
Retired

**Reply by TheCodeTraveler** [reply]
Here's two recommendations I found:
- Only use .NET NuGet Packages targeting .NET Core 3.1 (don't use .NET 5 dependencies)
  - https://github.com/Azure/azure-functions-core-tools/issues/2304#issuecomment-741534007
- In the Functions CSPROJ, add `<_FunctionsSkipCleanOutput>true</_FunctionsSkipCleanOutput>`
  - https://github.com/Azure/Azure-Functions/issues/1525#issuecomment-598584159
